From 128b4dcb34079718522bddedcfea323c1d44ca70 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@apache.org>
Date: Tue, 16 Sep 2014 21:29:09 -0700
Subject: [PATCH 405/791] HADOOP-11099. KMS return HTTP UNAUTHORIZED 401 on
 ACL failure. (tucu)

(cherry picked from commit e4ddb6da15420d5c13ec7ec99fed1e44b32290b0)

Conflicts:
	hadoop-common-project/hadoop-common/CHANGES.txt
---
 .../key/kms/server/KMSExceptionsProvider.java      |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
index 77b78ee..5cb0885 100644
--- a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
+++ b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
@@ -79,7 +79,7 @@ public Response toResponse(Exception exception) {
       // we don't audit here because we did it already when checking access
       doAudit = false;
     } else if (throwable instanceof AuthorizationException) {
-      status = Response.Status.UNAUTHORIZED;
+      status = Response.Status.FORBIDDEN;
       // we don't audit here because we did it already when checking access
       doAudit = false;
     } else if (throwable instanceof AccessControlException) {
-- 
1.7.9.5

